cpp/basix/lagrange.cpp

```diff
@@ -103,128 +103,61 @@ FiniteElement basix::create_lagrange(cell::type celltype, int degree)
     }
   }
 
-  std::size_t transform_count = 0;
-  for (std::size_t i = 1; i < topology.size() - 1; ++i)
-    transform_count += topology[i].size() * i;
-  auto base_transformations
-      = xt::tile(xt::expand_dims(xt::eye<double>(ndofs), 0), transform_count);
-  switch (celltype)
-  {
-  case cell::type::interval:
-    assert(transform_count == 0);
-    break;
-  case cell::type::triangle:
-  {
-    const std::vector<int> edge_ref
-        = doftransforms::interval_reflection(degree - 1);
-    for (int edge = 0; edge < 3; ++edge)
-    {
-      auto bt = xt::view(base_transformations, edge, xt::all(), xt::all());
-      const int start = 3 + edge_ref.size() * edge;
-      for (std::size_t i = 0; i < edge_ref.size(); ++i)
-      {
-        bt(start + i, start + i) = 0;
-        bt(start + i, start + edge_ref[i]) = 1;
-      }
-    }
-    break;
-  }
-  case cell::type::quadrilateral:
+  std::vector<xt::xtensor<double, 2>> entity_transformations;
+  if (tdim > 1)
   {
     const std::vector<int> edge_ref
         = doftransforms::interval_reflection(degree - 1);
-    for (int edge = 0; edge < 4; ++edge)
-    {
-      auto bt = xt::view(base_transformations, edge, xt::all(), xt::all());
-      const int start = 4 + edge_ref.size() * edge;
-      for (std::size_t i = 0; i < edge_ref.size(); ++i)
-      {
-        bt(start + i, start + i) = 0;
-        bt(start + i, start + edge_ref[i]) = 1;
-      }
-    }
-    break;
+    xt::xtensor<double, 2> et
+        = xt::zeros<double>({edge_ref.size(), edge_ref.size()});
+    for (std::size_t i = 0; i < edge_ref.size(); ++i)
```

Comment:
I think something like
```cpp
xt::view(et, xt::all(), xt::keep(edge_ref)) = 1;
```
should work here.

Reply:
amazing

Reply:
`xt::view(et, xt::all(), xt::keep(edge_ref))` gets the full block, not just the few entries we want. Can't immediately see a way to do this with a view